test/api/workflows_test.exs

```diff
@@ -13,291 +16,416 @@ defmodule StepFlow.Api.WorkflowsTest do
+          }
+        ]
       })
-      |> Router.call(@opts)
-      |> sent_resp
-
-    assert status == 201
-  end
 
-  test "SHOW /workflows/:id" do
-    {status, _headers, body} =
-      conn(:post, "/workflows", %{
-        identifier: "9A9F48E4-5585-4E8E-9199-CEFECF85CE14",
-        reference: "9A9F48E4-5585-4E8E-9199-CEFECF85CE14",
-        version_major: 1,
-        version_minor: 2,
-        version_micro: 3
-      })
-      |> Router.call(@opts)
-      |> sent_resp
-
-    assert status == 201
-
-    workflow_id =
-      body
-      |> Jason.decode!()
-      |> Map.get("data")
-      |> Map.get("id")
-      |> Integer.to_string()
-
-    {status, _headers, body} =
-      conn(:get, "/workflows/" <> workflow_id)
-      |> Router.call(@opts)
-      |> sent_resp
-
-    assert status == 200
-
-    data =
-      body
-      |> Jason.decode!()
-      |> Map.get("data")
-
-    identifier =
-      data
-      |> Map.get("identifier")
-
-    assert identifier == "9A9F48E4-5585-4E8E-9199-CEFECF85CE14"
-
-    reference =
-      data
-      |> Map.get("reference")
-
-    assert reference == "9A9F48E4-5585-4E8E-9199-CEFECF85CE14"
-  end
-
-  test "UPDATE /workflows/:id" do
-    {status, _headers, body} =
-      conn(:post, "/workflows", %{
-        identifier: "9A9F48E4-5585-4E8E-9199-CEFECF85CE14",
-        reference: "9A9F48E4-5585-4E8E-9199-CEFECF85CE14",
-        version_major: 1,
-        version_minor: 2,
-        version_micro: 3
-      })
-      |> Router.call(@opts)
-      |> sent_resp
-
-    assert status == 201
-
-    workflow_id =
-      body
-      |> Jason.decode!()
-      |> Map.get("data")
-      |> Map.get("id")
-      |> Integer.to_string()
-
-    {status, _headers, body} =
-      conn(:put, "/workflows/" <> workflow_id, %{workflow: %{reference: "updated reference"}})
-      |> Router.call(@opts)
-      |> sent_resp
-
-    assert status == 200
-
-    reference =
-      body
-      |> Jason.decode!()
-      |> Map.get("data")
-      |> Map.get("reference")
-
-    assert reference == "updated reference"
-  end
+      {status, _headers, body} =
+        conn(:get, "/workflows")
+        |> assign(:current_user, @unauthorized_user)
+        |> Router.call(@opts)
+        |> sent_resp
+
+      assert status == 200
+      assert body |> Jason.decode!() == %{"data" => [], "total" => 0}
+    end
+
+    @tag capture_log: true
+    test "POST /workflows valid with authorized user" do
+      {status, _headers, body} =
+        conn(:post, "/workflows", %{
+          workflow_identifier: "simple_workflow",
+          reference: "9A9F48E4-5585-4E8E-9199-CEFECF85CE14",
+          parameters: %{}
+        })
+        |> assign(:current_user, %{rights: ["user_create"]})
+        |> Router.call(@opts)
+        |> sent_resp
+
+      assert status == 201
+
+      assert body
+             |> Jason.decode!()
+             |> Map.get("data")
+             |> Map.get("reference") == "9A9F48E4-5585-4E8E-9199-CEFECF85CE14"
```

Comment:
Is it possible to right (not tested...):
```elixir
body
|> Jason.decode!()
|> Map.get("data")
|> Map.get("reference") == "9A9F48E4-5585-4E8E-9199-CEFECF85CE14"
|> assert

Reply:
warning: use of operator == has no effect
  test/api/workflows_test.exs:135